lib/fluent/supervisor.rb

```diff
@@ -302,6 +322,37 @@ def supervisor_sigusr2_handler
       $log.error "Failed to reload config file: #{e}"
     end
 
+    def supervisor_sigcont_handler
+      if Fluent.windows?
+        $log.info "dump file. [pid:#{Process.pid}]"
+
+        # Sigdump outputs under `/tmp` dir without `SIGDUMP_PATH` specified
+        # but `/tmp` dir may not exist on Windows by default.
+        # Since this function is mainly for Windows, this case should be saved.
+        # (Don't want to couple tightly with the Sigdump library but want to save this case especially.)
+        unless ENV['SIGDUMP_PATH']
+          dump_dir = '/tmp'
+          unless Dir.exist?(dump_dir)
+            FileUtils.mkdir_p(dump_dir, mode: Fluent::DEFAULT_DIR_PERMISSION)
```

Comment:
How about using OS's temp directory by default? (`%windir%\Temp`)

Reply:
I have fixed it on https://github.com/fluent/fluentd/pull/3680/commits/65407e59a5200e21d1449815b4a310073cf27976.

In addition to using `%windir%/Temp` by default, now add pid to the path specified by `SIGDUMP_PATH` environment variable on Windows in order to avoid the writing conflicting.

Example.
```powershell
$ $env:SIGDUMP_PATH="/sigdump/sigdump.log"
$ fluentd -c "{conf}"
(sigdump action)
```

```
... [info]: fluent/log.rb:330:info: dump to /sigdump/sigdump-41544.log.
... [info]: #0 fluent/log.rb:330:info: dump to /sigdump/sigdump-21152.log.
... [info]: #1 fluent/log.rb:330:info: dump to /sigdump/sigdump-15656.log.
```

As for UNIX-like, in order to add the same specification, 
need to add a trap of SIGCONT to the worker processes
or to fix the Sigdump library.
However, it loses backward compatibility for users using 'SIGDUMP_PATH'
environment variable, so need to consider what to do.

Reply:
For UNIX-like, it would be better to leave it as is to keep compatibility.
On the other hand, it would be also better that modifying sigdump for specifying a directory.
In current implementation, sigdump assumes the path specified by `SIGDUMP_PATH` is a file.

https://github.com/fluent/sigdump/blob/c9fcf34ee58b529cb46c528b700f4ff491326d4a/lib/sigdump.rb#L134-L148

```ruby
  def self._open_dump_path(path, &block)
    case path
    when nil, ""
      path = "/tmp/sigdump-#{Process.pid}.log"
      File.open(path, "a", &block)
    when IO
      yield path
    when "-"
      yield STDOUT
    when "+"
      yield STDERR
    else
      File.open(path, "a", &block)
    end
  end
```

But when the path is ended with `/` or the patch exists and it's directory, it should be treated as directory.
When a directory is specified, it would be better that sigdump adds pid automatically.

Reply:
Thanks for your advice.

> For UNIX-like, it would be better to leave it as is to keep compatibility.

Then the best solution is 

- Abolish trapping SIGCONT in the supervisor process on UNIX-like.
  - Don't change SIGCONT behavior on UNIX-like for compatibility.
- `/api/processes.dump` rpc command on UNIX-like sends SIGCONT to the supervisor process and all worker processes.
  - This is a new function, so we don't have to care about compatibility. 
  - With `SIGDUMP_PATH` specified, the writing confliction will occur. This problem should be fixed by improving the Sigdump library so that we can specify a directory path.

`/api/processes.dump` for UNIX-like
```ruby
Process.kill :CONT, $$
send_signal_to_workers(:CONT)
```

Reply:
I have fixed this on https://github.com/fluent/fluentd/pull/3680/commits/9d0d1f331dda7a10d76a102b514096719cb8248e.

- SIGCONT behavior on UNIX-like remains as before.
- `/api/processes.dump` rpc command on UNIX-like sends SIGCONT to the supervisor process and all worker processes.
- Fix the Sigdump library later. Until then, we cannot use this rpc command with `SIGDUMP_PATH` on UNIX-like.